airbyte-webapp/src/components/CreateConnectionContent/CreateConnectionContent.tsx

```diff
@@ -31,61 +31,49 @@ const CreateConnectionContent: React.FC<CreateConnectionContentProps> = ({
   afterSubmitConnection,
 }) => {
   const { mutateAsync: createConnection } = useCreateConnection();
-  const analyticsService = useAnalyticsService();
+  const { push } = useRouter();
+
+  const formId = useUniqueFormId();
+  const { clearFormChange } = useFormChangeTrackerService();
 
   const { schema, isLoading, schemaErrorStatus, catalogId, onDiscoverSchema } = useDiscoverSchema(
     source.sourceId,
     true
   );
 
-  const connection = useMemo<ConnectionFormProps["connection"]>(
-    () => ({
-      syncCatalog: schema,
-      destination,
-      source,
-      catalogId,
-    }),
-    [schema, destination, source, catalogId]
-  );
-
-  const onSubmitConnectionStep = async (values: ValuesProps) => {
-    const connection = await createConnection({
-      values,
-      source,
-      destination,
-      sourceDefinition: {
-        sourceDefinitionId: source?.sourceDefinitionId ?? "",
-      },
-      destinationDefinition: {
-        name: destination?.name ?? "",
-        destinationDefinitionId: destination?.destinationDefinitionId ?? "",
-      },
-      sourceCatalogId: catalogId,
-    });
-
-    return {
-      onSubmitComplete: () => {
-        afterSubmitConnection?.(connection);
-      },
-    };
+  const connection = {
+    syncCatalog: schema,
+    destination,
+    source,
+    catalogId,
   };
 
-  const onSelectFrequency = (item: IDataItem | null) => {
-    const enabledStreams = connection.syncCatalog.streams.filter((stream) => stream.config?.selected).length;
-
-    if (item) {
-      analyticsService.track(Namespace.CONNECTION, Action.FREQUENCY, {
-        actionDescription: "Frequency selected",
-        frequency: item.label,
-        connector_source_definition: source?.sourceName,
-        connector_source_definition_id: source?.sourceDefinitionId,
-        connector_destination_definition: destination?.destinationName,
-        connector_destination_definition_id: destination?.destinationDefinitionId,
-        available_streams: connection.syncCatalog.streams.length,
-        enabled_streams: enabledStreams,
+  const onSubmitConnectionStep = useCallback(
+    async (values: ValuesProps) => {
+      const createdConnection = await createConnection({
+        values,
+        source,
+        destination,
+        sourceDefinition: {
+          sourceDefinitionId: source?.sourceDefinitionId ?? "",
+        },
+        destinationDefinition: {
+          name: destination?.name ?? "",
+          destinationDefinitionId: destination?.destinationDefinitionId ?? "",
+        },
+        sourceCatalogId: catalogId,
       });
-    }
-  };
+
+      // We only want to go to the new connection if we _do not_ have an after submit action.
+      if (!afterSubmitConnection) {
+        // We have to clear the form change to prevent the dirty-form tracking modal from appearing.
+        clearFormChange(formId);
```

Comment:
This looks too high in the component tree and should be called from the Connection form. The reason is that the ConnectionForm sets up the ConnectionChangeTracker, and should be responsible for clearing it after submission. In this case, this component, which has no idea what the form does, assumes that the tracker is being used.

Reply:
The next PR brings the FormikContext "out" a layer at which point this would be the right place for this call.  Does it seem reasonable to leave this for now knowing it will be much more readily adressable in the next PR?

Reply:
This will be changed in part 2.